src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -684,6 +684,13 @@ boost::shared_ptr<ConsoleProcess> ConsoleProcess::create(
       InteractionMode interactionMode,
       int maxOutputLines)
 {
+   // return existing ConsoleProcess
+   ProcTable::const_iterator pos = s_procs.find(terminalHandle);
+   if (pos != s_procs.end())
+   {
+      return boost::shared_ptr<ConsoleProcess> (pos->second);
```

Comment:
`pos->second` is already a `boost::shared_ptr` -- do we need to construct the temporary here?